jmespath-core/src/main/java/io/burt/jmespath/Adapter.java

```diff
@@ -82,6 +94,12 @@
   T getProperty(T value, String name);
 
   /**
+   * Returns all of the property names of the given object, or an empty list
+   * when the given value is not an object.
+   */
+  Collection<String> getPropertyNames(T value);
```

Comment:
The `List` -> `Collection` change was something I changed on a whim, it's kind of the same argument as `Double` vs. `Number`. I'm not sure yet exactly how specific this API should be.